src/utils/telemetry/tracked-command.js

```diff
@@ -0,0 +1,51 @@
+const { hrtime } = require('process')
+
+const { Command } = require('@oclif/command')
+const { ExitError } = require('@oclif/errors')
+
+const { track } = require('.')
+
+const NANO_SECS_TO_MSECS = 1e6
+
+const getDuration = function (startTime) {
+  const durationNs = hrtime.bigint() - startTime
+  return Math.round(Number(durationNs / BigInt(NANO_SECS_TO_MSECS)))
+}
+
+const isSafeError = function (error) {
+  // The `exit` oclif utility returns an ExitError even when the exit code is 0
+  return error instanceof ExitError && error.oclif && error.oclif.exit === 0
+}
+
+const getStatus = function (error) {
+  if (error === undefined || isSafeError(error)) {
+    return 'success'
+  }
+
+  return 'error'
+}
+
+class TrackedCommand extends Command {
+  async init() {
+    this.analytics = { startTime: hrtime.bigint() }
```

Comment:
We could do this as a class properly to avoid someone extending and forgetting to call `super.init`.

However, based on [this code](https://github.com/oclif/command/blob/adc0b266777b515a26365d31c0e3a926b68d0c72/src/command.ts#L145) I think `init` should always be called, and it seems the correct place to start measuring command time.

We could also move it to `run` but then we would have to call `super.run` in every command which doesn't feel correct.